File: 2419. Longest Subarray With Maximum Bitwise AND.py
class Solution {
    public int longestSubarray(int[] nums) {
        int len = 1;
         
        int max = Integer.MIN_VALUE;
        for (int num : nums){
            max = Math.max(max, num);
        } 

        int tmp_len = 0;
        for(int num: nums){
            if(num == max){
                tmp_len ++;
            }
            else{
                len = Math.max(len,tmp_len);
                tmp_len = 0;
            }
        }
        len = Math.max(len,tmp_len);
        return len;
    }
}

File: 2419. Longest Subarray With Maximum Bitwise AND.java
class Solution {
    public int longestSubarray(int[] nums) {
        int maxA=0;
        for(int n:nums)
        maxA=Math.max(maxA,n);
        int count=0;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]==maxA)
            {
                int c1=0;
                while(i<nums.length&&nums[i]==maxA)
                {
                    c1++;
                    i++;
                }
                count=Math.max(count,c1);
                i--;
            }
        }
        return count;
        
    }
}
